nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-framework-core/src/main/java/org/apache/nifi/controller/repository/FileSystemRepository.java

```diff
@@ -392,8 +399,11 @@ public long getContainerUsableSpace(String containerName) throws IOException {
         if (path == null) {
             throw new IllegalArgumentException("No container exists with name " + containerName);
         }
-
-        return Files.getFileStore(path).getUsableSpace();
+        long usableSpace=path.toFile().getUsableSpace();
+        if(usableSpace==0) {
+            throw new RuntimeException("System returned usable space of the partition for " + containerName + " is zero byte. Nifi can not create a zero sized FileSystemRepository");
```

Comment:
After more thought, I agree with the concerns that @markap14 raised.  I think the constructor can throw RuntimeException as it exists now.  But I think the getContainerCapacity() should throw IOException and getContainerUsableSpace() should just return the 0.